src/xunit.performance.api/ETWProfiler.cs

```diff
@@ -107,9 +116,142 @@ public static void Profile(string assemblyFileName, string sessionName, Action a
 
             TraceEventSession.MergeInPlace(userFullFileName, Console.Out);
             WriteInfoLine($"ETW Tracing Session saved to \"{userFullFileName}\"");
+            collectOutputFilesCallback(userFullFileName);
+
+            var assemblyModel = GetAssemblyModel(assemblyFileName, userFullFileName, sessionName, performanceTestMessages);
+            var fileNameWithoutExtension = Path.GetFileNameWithoutExtension(userFullFileName);
+            var xmlFileName = Path.Combine(outputDirectory, $"{fileNameWithoutExtension}.xml");
+            new AssemblyModelCollection { assemblyModel }.Serialize(xmlFileName);
+            WriteInfoLine($"Performance results saved to \"{xmlFileName}\"");
+            collectOutputFilesCallback(xmlFileName);
+
+            var mdFileName = Path.Combine(outputDirectory, $"{fileNameWithoutExtension}.md");
+            var dt = CreateStatistics(assemblyModel);
+            var mdTable = MarkdownHelper.GenerateMarkdownTable(dt);
+            MarkdownHelper.Write(mdFileName, mdTable);
+            WriteInfoLine($"Markdown file saved to \"{mdFileName}\"");
+            collectOutputFilesCallback(mdFileName);
+            Console.WriteLine(mdTable);
+
+            var csvFileName = Path.Combine(outputDirectory, $"{fileNameWithoutExtension}.csv");
+            dt.WriteToCSV(csvFileName);
+            WriteInfoLine($"Statistics written to \"{csvFileName}\"");
+            collectOutputFilesCallback(csvFileName);
+        }
+
+        private static DataTable CreateStatistics(AssemblyModel assemblyModel)
```

Comment:
I take it this is covered under the outstanding issue to clean-up the output generation code?  This looks like it belongs in a different file and not as part of the ETW profiling code.